prow/jobs/images/Dockerfile.integration

```diff
@@ -16,8 +16,8 @@ RUN echo "Installing Kustomize ..." \
     && rm "${KUSTOMIZE_TARBALL}"
 
 RUN echo "Installing Kubectl ... " \
-    && curl -fsSLo /usr/share/keyrings/kubernetes-archive-keyring.gpg https://packages.cloud.google.com/apt/doc/apt-key.gpg \
-    && echo "deb [signed-by=/usr/share/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | tee /etc/apt/sources.list.d/kubernetes.list \
+    && curl -fsSL https://packages.cloud.google.com/apt/doc/apt-key.gpg | gpg --yes --dearmor -o /usr/share/keyrings/kubernetes-archive-keyring.gpg \
```

Comment:
Thinking about moving to something like `   curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"`

Reply:
https://github.com/kubernetes/release/issues/2860 for context